analysis/03_new_model_training/06_grid_To_mun_with_new_weightFile.md

```diff
@@ -0,0 +1,453 @@
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X_scaled,
+    df_data["percent_buildings_damaged"],
+    stratify=y_input_strat,
+    test_size=0.2,
+)
+
+# Define XGBoost Reduced Overfitting model
+xgb = XGBRegressor(
+    base_score=0.5,
+    booster="gbtree",
+    colsample_bylevel=0.8,
+    colsample_bynode=0.8,
+    colsample_bytree=0.8,
+    gamma=3,
+    eta=0.01,
+    importance_type="gain",
+    learning_rate=0.1,
+    max_delta_step=0,
+    max_depth=4,
+    min_child_weight=1,
+    missing=1,
+    n_estimators=100,
+    early_stopping_rounds=10,
+    n_jobs=1,
+    nthread=None,
+    objective="reg:squarederror",
+    reg_alpha=0,
+    reg_lambda=1,
+    scale_pos_weight=1,
+    seed=None,
+    silent=None,
+    subsample=0.8,
+    verbosity=1,
+    eval_metric=["rmse", "logloss"],
+    random_state=0,
+)
+
+eval_set = [(X_test, y_test)]
+xgb_model = xgb.fit(X_train, y_train, eval_set=eval_set, verbose=False)
+
+X2 = sm.add_constant(X_train)
+est = sm.OLS(y_train, X2)
+est2 = est.fit()
+print(est2.summary())
+```
+
+```python
+# Combine test and training set to have predictiob for all
+y_pred_all = xgb.predict(X_scaled)
+pred_df = pd.DataFrame(columns=["y_all", "y_pred_all"])
+pred_df["y_all"] = df_data["percent_buildings_damaged"]
+pred_df["y_pred_all"] = y_pred_all
+
+pred_df
+```
+
+```python
+rmse = sqrt(mean_squared_error(pred_df["y_all"], pred_df["y_pred_all"]))
+print(f"RMSE for grid_based model: {rmse:.2f}")
+```
+
+```python
+bin_index = np.digitize(pred_df["y_all"], bins=binsP2)
+```
+
+```python
+# Join data with y_all and y_all_pred
+df_data_w_pred = pd.merge(pred_df, df_data, left_index=True, right_index=True)
+# Join data with grid_point_id typhoon_year
+df_data_w_pred_grid = pd.merge(
+    df[["grid_point_id", "typhoon_year"]],
+    df_data_w_pred,
+    left_index=True,
+    right_index=True,
+)
+```
+
+```python
+df_data_w_pred_grid.sort_values("y_pred_all", ascending=False)
+```
+
+```python
+# join with weights df
+join_df = df_data_w_pred_grid.merge(df_weight, on="grid_point_id", how="left")
+join_df
+```
+
+```python
+# Indicate where values are valid and not missing
+join_df = join_df.loc[join_df["weight"].notna()]
+join_df
+```
+
+```python
+# Multiply weight by y_all and y_pred_all
+join_df["weight*y_pred*buildings"] = (
+    join_df["y_pred_all"] * join_df["weight"] * join_df["total_buildings"] / 100
+)
+join_df["weight*y*buildings"] = (
+    join_df["y_all"] * join_df["weight"] * join_df["total_buildings"] / 100
+)
+join_df["weight*buildings"] = join_df["weight"] * join_df["total_buildings"]
+join_df
+```
+
+```python
+join_df.sort_values("y_pred_all", ascending=False)
+```
+
+```python
+# Groupby by municipality and typhoon_name with sum as the aggregation function
+agg_df = join_df.groupby(["ADM3_PCODE", "typhoon_name", "typhoon_year"]).agg("sum")
+
+# Normalize by the sum of the weights
+agg_df["y_pred_norm"] = (
+    agg_df["weight*y_pred*buildings"] / agg_df["weight*buildings"] * 100
```

Comment:
I think this is where the main problem lies: to compare with the municipality based model, the denominator should be the same that Aki used to get the percentages. Or, you could just compare the total number of buildings and not do any normalization. Either way, you'll need to compute the denominator that Aki used, as was done in [this notebook](https://github.com/rodekruis/Typhoon-Impact-based-forecasting-model/blob/master/IBF-Typhoon-model/documentation/notebooks/pre_disaster_indicators.ipynb)

Reply:
ok, if we do not divide by weight*buildings we should get the number of damaged buildings per municiplity

Reply:
After [this](https://humdata.slack.com/archives/C03H9QCBGB0/p1674584983403429) discussion with Marc and Aki, it looks like we should really be normalizing our building dataset in the same way as the municipality-based model. Hopefully this will resolve the issue.